openfeature/_event_support.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from typing import TYPE_CHECKING, Dict, List
+
+from openfeature.event import (
+    EventDetails,
+    EventHandler,
+    ProviderEvent,
+    ProviderEventDetails,
+)
+from openfeature.provider import FeatureProvider
+
+if TYPE_CHECKING:
+    from openfeature.client import OpenFeatureClient
+
+
+_global_handlers: Dict[ProviderEvent, List[EventHandler]] = defaultdict(list)
+_client_handlers: Dict[
+    OpenFeatureClient, Dict[ProviderEvent, List[EventHandler]]
+] = defaultdict(lambda: defaultdict(list))
+
+
+def run_client_handlers(
+    client: OpenFeatureClient, event: ProviderEvent, details: EventDetails
+) -> None:
+    for handler in _client_handlers[client][event]:
+        handler(details)
+
+
+def run_global_handlers(event: ProviderEvent, details: EventDetails) -> None:
+    for handler in _global_handlers[event]:
+        handler(details)
+
+
+def add_client_handler(
+    client: OpenFeatureClient, event: ProviderEvent, handler: EventHandler
+) -> None:
+    handlers = _client_handlers[client][event]
+    handlers.append(handler)
+
+    _run_immediate_handler(client, event, handler)
+
+
+def remove_client_handler(
+    client: OpenFeatureClient, event: ProviderEvent, handler: EventHandler
+) -> None:
+    handlers = _client_handlers[client][event]
+    handlers.remove(handler)
```

Comment:
do we want to be more forgiving when something is tried to be removed, but doesn't exist anymore?